Xamarin.Forms.Core/Page.cs

```diff
@@ -421,5 +426,16 @@ bool ShouldLayoutChildren()
 		{
 			return _platformConfigurationRegistry.Value.On<T>();
 		}
+
+		internal void SetTitleView(View oldTitleView, View newTitleView)
+		{
+			if (oldTitleView != null)
+				oldTitleView.Parent = null;
+
+			if (newTitleView != null)
+				newTitleView.Parent = this;
```

Comment:
does Parenting ensure that the InheritedBindingContext is set ? if so, 👍 , we're good

Reply:
Setting the parent ensures that it gets set initially but it doesn't create a link that causes it to get updated in the future.  

The typical pattern around the code is for the Parent to know about the children and then propagate the BindingContext by overriding *BindingContextChanged*. That's how Label propagates to Span, VisualElements propagate to Gestures, and Spans propagate to gestures.